File: Retrofit/Java/RetrofitNewExample/app/src/main/java/com/example/retrofitnewexample/MainActivity.java
package com.example.retrofitnewexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    TextView theText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        theText = (TextView)findViewById(R.id.thetext);

        //Instância da interface de serviços
        RetrofitServices services = RetrofitClientInstance.getRetrofit().create(RetrofitServices.class);

        //Pegando os "usuários" da página e colocando em uma call
        Call<List<User>> call = services.getUsers();

        //O método abaixo "pega" os elementos da call um por um, e para cada um deles há um
        //"OnResponse" caso responda com sucesso e um "OnFailure" caso haja falhas

        call.enqueue(new Callback<List<User>>() {
            @Override
            public void onResponse(Call<List<User>> call, Response<List<User>> response) {
                //Caso haja sucesso

                //Pegue a lista de "Users" da página

                List<User> users = response.body();

                for(User us: users){

                    String textToAppend = "";

                    textToAppend += "userId: "+us.getUserId()+"\n";
                    textToAppend += "Id: "+us.getId()+"\n";
                    textToAppend += "Title: "+us.getTitle()+"\n";
                    textToAppend += "Text: "+us.getBody()+"\n";
                    textToAppend += "\n";

                    theText.append(textToAppend);

                }

            }

            @Override
            public void onFailure(Call<List<User>> call, Throwable t) {
                theText.setText(t.getMessage());
            }
        });
    }
}

File: Retrofit/Kotlin/KotlinRetrofitExample/app/src/main/java/com/example/kotlinretrofitexample/MainActivity.kt
package com.example.kotlinretrofitexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val call = RetrofitInitialization().theServices().getUsers()

        call.enqueue(object : Callback<List<User>>{
            override fun onFailure(call: Call<List<User>>, t: Throwable) {

            }

            override fun onResponse(call: Call<List<User>>, response: Response<List<User>>) {

                var users : List<User>? = response.body()

                for(user in users!!){

                    var textToAppend : String = ""

                    textToAppend += "userId: " + user.userId + "\n"
                    textToAppend += "Id: " + user.id + "\n"
                    textToAppend += "Title: " + user.title + "\n"
                    textToAppend += "Text: " + user.body + "\n"
                    textToAppend += "\n"

                    thetext.append(textToAppend)

                }

            }


        })
    }
}
